contrib/scalapblib/src/ScalaPBWorker.scala

```diff
@@ -25,17 +25,18 @@ class ScalaPBWorker {
 
         val instance = new ScalaPBWorkerApi {
           override def compileScalaPB(
-              source: File,
+              root: File,
+              sources: Seq[File],
               scalaPBOptions: String,
               generatedDirectory: File,
               otherArgs: Seq[String]
           ): Unit = {
             val opts = if (scalaPBOptions.isEmpty) "" else scalaPBOptions + ":"
             val args = otherArgs ++ Seq(
               s"--scala_out=${opts}${generatedDirectory.getCanonicalPath}",
-              s"--proto_path=${source.getParentFile.getCanonicalPath}",
-              source.getCanonicalPath
-            )
+              s"--proto_path=${root.getCanonicalPath}"
+            ) ++ sources.map(_.getCanonicalPath)
+            ctx.log.debug(args.mkString(" "))
```

Comment:
I would prefer to have a more meaningful message, e.g. 

```suggestion
            ctx.log.debug("ScalaPBC args: ${args.mkString(" ")}")
```

Reply:
Good idea thank you for the suggestion  i have added it.